cms/djangoapps/contentstore/views/tests/test_group_configurations.py

```diff
@@ -675,6 +687,54 @@ def test_can_get_correct_usage_info_for_content_groups(self):
 
         self.assertEqual(actual, expected)
 
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_can_get_correct_usage_info_with_orphan(self, module_store_type):
+        """
+        Test if content group json updated successfully with usage information even if there is
+        an orphan in content group.
+        """
+        course = CourseFactory.create(default_store=module_store_type)
+
+        self._add_user_partitions(course=course, count=1, scheme_id='cohort')
+
+        vertical, problem = self._create_problem_with_content_group(cid=0, group_id=1, course=course, name_suffix='0')
+
+        # Assert that there is no orphan in the course yet.
+        self.assertEqual(len(self.store.get_orphans(course.id)), 0)
+
+        # Get the content group information
+        actual = GroupConfiguration.get_or_create_content_group(self.store, course)
+
+        # Get expected content group for the added problem.
+        expected = self._get_expected_content_group(usage_for_group=[
+            {
+                'url': '/container/{}'.format(vertical.location),
+                'label': 'Test Unit 0 / Test Problem 0'
+            }
+        ])
+
+        # Assert that actual content group information is same as expected one.
+        self.assertEqual(actual, expected)
+
+        # Update problem(created earlier) to an orphan.
+        with self.store.branch_setting(ModuleStoreEnum.Branch.published_only):
+            vertical = self.store.get_item(vertical.location)
+            vertical.children.remove(problem.location)
+            self.store.update_item(vertical, self.user.id)
+
+        # Assert that the vertical is orphan now.
+        self.assertIn(problem.location, self.store.get_orphans(course.id))
+
+        # Get the content group information
+        actual = GroupConfiguration.get_or_create_content_group(self.store, course)
+
+        # For mongo modulestore expected content group usage information will NOT change.
+        if module_store_type == ModuleStoreEnum.Type.split:
+            expected = self._get_expected_content_group(usage_for_group=[])
```

Comment:
instead of having conditionals in the test, can you include this information in the ddt?